crates/router/src/core/admin.rs

```diff
@@ -2916,6 +2944,47 @@ pub async fn retrieve_connector(
     ))
 }
 
+#[cfg(feature = "v2")]
+pub async fn list_connectors_for_a_profile(
+    state: SessionState,
+    merchant_id: id_type::MerchantId,
```

Comment:
Instead of passing merchant_id pass merchant_account, read merchant account in auth/middleware layer itself.